### PR TITLE
HWKAPM-737 Fix issue where 'All' transaction (when other txns defined…

### PR DESCRIPTION
**Pull request by objectiser**
…) does not retrieve the list of hostnames due to passing empty string in criteria.transaction field

### CODE REVIEW

**Comment by objectiser** [comment]
@ammendonca Could you review please?

This is a quick fix as currently only the hostnames query seems to be affected, but I've created a new bug (HWKAPM-741) as the criteria semantics is not being used consistently across the other queries - so eventually this fix would need to be applied to all queries that are supplying criteria. So if there is a more general solution it could be applied later as part of that issue?

File: ui/src/main/scripts/plugins/directives/filter-sidebar/ts/filterSidebarDirective.ts [hunk 1]
```diff
@@ -165,5 +165,12 @@ module FilterSidebar {
 
     }
 
+    private criteriaToJson(criteria) {
+      let txncriteria = angular.copy(criteria);
+      if (txncriteria.transaction === '') {
+        delete txncriteria['transaction'];
```
Comment (ammendonca) [inline ui/src/main/scripts/plugins/directives/filter-sidebar/ts/filterSidebarDirective.ts]: I would just use dot notation, so `delete txncriteria.transaction;`.

**Comment by objectiser** [comment]
@ammendonca Thanks, update applied. Will merge once CI finished.

**Comment by objectiser** [comment]
Merging, as jenkins failure not related to this PR.

**Pull request merged**